packages/middleware-endpoint/src/service-customizations/s3.ts

```diff
@@ -26,3 +45,13 @@ export const S3_HOSTNAME_PATTERN = /^(.+\.)?s3(-fips)?(\.dualstack)?[.-]([a-z0-9
  */
 export const isDnsCompatibleBucketName = (bucketName: string): boolean =>
   DOMAIN_PATTERN.test(bucketName) && !IP_ADDRESS_PATTERN.test(bucketName) && !DOTS_PATTERN.test(bucketName);
+
+export const isArnBucketName = (bucketName: string): boolean => {
+  const [arn, partition, service, region, account, typeOrId] = bucketName.split(":");
+  const isArn = arn === "arn" && bucketName.split(":").length >= 6;
+  const isValidArn = [arn, partition, service, account, typeOrId].filter(Boolean).length === 5;
+  if (isArn && !isValidArn) {
+    throw new Error(`Invalid ARN: ${bucketName} was an invalid ARN.`);
+  }
+  return arn === "arn" && !!partition && !!service && !!account && !!typeOrId;
+};
```

Comment:
I would expect that we already have such a function somewhere else, but I couldn't find one.